### PR TITLE
Add headPipeline to MergeRequest

### PR DESCRIPTION
**Pull request by vimsucks**
[https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr)

image

image

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/models/MergeRequest.java [hunk 1]
```diff
@@ -329,6 +330,15 @@ public void setPipeline(Pipeline pipeline) {
...
     }
 
+    public Pipeline getHeadPipeline() {
+        return headPipeline;
+    }
+
+    public MergeRequest setHeadPipeline(Pipeline headPipeline) {
+        this.headPipeline = headPipeline;
+        return this;
+    }
```
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/MergeRequest.java:340]: In this project the setters are returning `void`.

When an attribute is used to create or update an entity, there is a builder method `withHeadPipeline(..)` next to the setter.
But in this case the `head_pipeline` attribute is only used during a read, so the builder method makes no sense.

```suggestion
    public void setHeadPipeline(Pipeline headPipeline) {
        this.headPipeline = headPipeline;
    }
```

File: src/main/java/org/gitlab4j/api/models/Pipeline.java [hunk 2]
```diff
@@ -33,6 +34,15 @@ public void setId(Long id) {
...
     }
 
+    public Long getProjectId() {
+        return projectId;
+    }
+
+    public Pipeline setProjectId(Long projectId) {
+        this.projectId = projectId;
+        return this;
+    }
```
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/Pipeline.java:44]: Same here:

```suggestion
    public void setProjectId(Long projectId) {
        this.projectId = projectId;
    }
```

**Review by jmini — CHANGES_REQUESTED** [review]
I would do minor changes to be consistent with the rest of the code. But for me this change looks good to me.

**Comment by vimsucks** [comment]
> I would do minor changes to be consistent with the rest of the code. But for me this change looks good to me.

thanks! suggestion applied

**Review by jmini — APPROVED** [review]

**Pull request merged**